### PR TITLE
test(worker): refactor worker tests

### PR DESCRIPTION
**Pull request by jerome-benoit**
### What does this PR do?

+ constify variables
+ typing
+ untangle web worker from worker_threads tests
+ cleanup variables namespace
+ move worker related tests into workers directory

no functional changes on tests

### How did you verify your code works?

ran impacted tests

### CODE REVIEW

**Comment by jerome-benoit** [comment]
@Electroid: May I ask for your review on that PR? 
The repo setup does not allow me to ask for a review via the intended field. 

I really would like to have the worker tests refactoring one merged. I have other fixes to `worker_threads` code stacked on top of that PR that I would like to push upstream.  

I also have a question regarding design discussions: where do they usually happen? 
I need inputs by people having deep bun internals knowledge to tackle properly some of the issues encountered with `worker_threads`.

File: test/js/web/worker-fixture-env.js
Comment (paperclover) [inline test/js/web/worker-fixture-env.js:3]: this check was removed

File: test/js/node/worker_threads/worker.js
Comment (paperclover) [inline test/js/node/worker_threads/worker.js]: `?.` -> `.`

we want an error to be thrown if this property is undefined.

**Review by paperclover — CHANGES_REQUESTED** [review]
thanks for cleaning up the mess that is these worker tests.

two nitpicks i would like fixed before merging, see review comments.

also, if you would like to add this, could you add an extra test that checks if a `.tsx` file works. theyre supposed to but re-reading these tests there isnt any coverage of that. no worries if not.

File: test/js/web/worker-fixture-env.js
Comment (jerome-benoit) [reply test/js/web/worker-fixture-env.js:3]: You mean the test has been removed upstream? 
I've used `git mv` twice on few files and dunno if it's a git bug or expected behavior but the link to the original file and its changes is broken for these files.
I will inspect upstream changes, address review comments then ask for review in a few days.

Comment (paperclover) [reply test/js/web/worker-fixture-env.js:3]: it was deleted here where i commented and not re-added in `test/js/web/workers/worker-fixture-env.js` where it was moved.

File: test/js/node/worker_threads/worker.js
Comment (jerome-benoit) [reply test/js/node/worker_threads/worker.js]: TS complains about 'parentPort can be null'. So I've added a null check + error throwing in `node:worker_threads` related workers.

File: test/js/web/worker-fixture-env.js
Comment (jerome-benoit) [reply test/js/web/worker-fixture-env.js:3]: The check has been moved to `node:worker_threads` tests worker implementation because `isMainThread` is not part of the web worker API.

Tests entanglement between worker_threads and web worker is not complete yet.

I will do another PR to move worker_threads tests and workers code to node:worker_threads test directory.

**Comment by jerome-benoit** [comment]
> also, if you would like to add this, could you add an extra test that checks if a `.tsx` file works. theyre supposed to but re-reading these tests there isnt any coverage of that. no worries if not.

As part of another PR then.

+ one PR to finish the worker_threads and web worker test code untanglement
+ one PR to add a test for `.tsx` worker file handling for web worker

File: test/js/node/worker_threads/worker.js
Comment (paperclover) [reply test/js/node/worker_threads/worker.js]: ah i see

File: test/js/web/worker-fixture-env.js
Comment (paperclover) [reply test/js/web/worker-fixture-env.js:3]: i guess this makes sense, but it is something bun allows, but i don't think we explicilty document this behavior anywhere.

**Review by paperclover — APPROVED** [review]
looks good. before i run the tests i am going to push a commit fixing the submodule. will merge these changes after tests run

**Pull request merged**
